### ISSUE TITLE
Add NullFriendlyCollections.astub stub file

### ISSUE THREAD

**Issue by mernst** [typetools/checker-framework#3040]
Consider a method such as `Map.remove()`.

Some versions of `Map` are null-hostile and will throw an exception if null is passed to `remove()`.
Other versions of `Map` are null-friendly and do not throw an exception if null is passed to `remove()`.

It is unsound to annotate `Map.remove()` as taking nullable arguments:
```
  default boolean remove(@Nullable Object key, @Nullable Object value) {
```
because code that passes the Nullness Checker may still throw a NullPointerException.
Another way of explaining why this is unsound is that `@Nullable` means, "No null pointer exception will be thrown due to null being passed as an argument.", and that is not true in this case.

We would like to annotate it more precisely and permissively than only accepting non-null arguments, however.  https://github.com/typetools/checker-framework/issues/1326 is an issue for this.

Until #1326 is resolved, a workaround is to have a stub file that assumes that all collections are null-friesdly.
This stub file can be used by people who wish to reduce the number of false positives when type-checking code with null-friendly collections, and do not care about the unsoundness.
